src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -165,6 +165,27 @@ const peerConnectionBrokenChangeHandler = async (
   dispatch(setToastMsg(ToastMsgType.DISCONNECT_WALLET_SUCCESS));
 };
 
+const removeConnectionsPendingDeletion = async (
+  dispatch: ReturnType<typeof useAppDispatch>
+) => {
+  const pendingDeletions =
+    await Agent.agent.connections.getConnectionsPendingDeletion();
+
+  for (const id of pendingDeletions) {
+    await Agent.agent.connections.deleteConnectionById(id);
+    dispatch(removeConnectionCache(id));
+  }
+};
+
+const resolvePendingConnections = async () => {
+  const pendingConnections =
+    await Agent.agent.connections.getConnectionsPending();
+
+  for (const pendingConnection of pendingConnections) {
+    await Agent.agent.connections.resolveOobi(pendingConnection.oobi);
+  }
+};
```

Comment:
Sorry I missed this in lat review but these functions don't below here since they are for core. We need to start thinking of UI and core as separate packages.

Can you move these to the connection service and call them from agent? (and please don't use `Agent.agent`, call the connection service object directly)

Reply:
@Sotatek-BaoHoanga What I mean is the core needs to do this itself, not the UI... Sorry, I got it wrong the first time.

What you need to do is to have these functions in the core, and call them in `markAgentStatus` if the status is ONLINE.

And in `removeConnectionsPendingDeletion`, whenever you delete a connection, you should emit an event to delete the connection from the UI/Redux. (Not sure if it already does this anywhere)